calico_kubernetes/calico_kubernetes.py

```diff
@@ -726,15 +728,55 @@ def run_protected():
     """
     Runs the plugin, intercepting all exceptions.
     """
-    # Configure logging
-    configure_logger(logger, LOG_LEVEL, True)
-    configure_logger(pycalico_logger, LOG_LEVEL, False)
+    # Parse arguments and configure logging
+    global logger
+    mode = sys.argv[1]
+
+    # Append a stdout logging handler to log to the Kubelet.
+    # We cannot do this in the status hook because the Kubelet looks to
+    # stdout for status results.
+    if mode != 'status':
+        configure_stdout_logger(logger=logger,
+                                log_level=logging.INFO,
+                                root_logger=True)
+        configure_stdout_logger(logger=pycalico_logger,
+                                log_level=logging.INFO,
+                                root_logger=False)
+
+    # Adapt the logger to append the Docker ID to logs.
+    # Docker ID is not supplied in `init`, so do not include the it in logs.
+    if mode != 'init':
+        namespace = sys.argv[2].replace('/', '_')
+        pod_name = sys.argv[3].replace('/', '_')
+        docker_id = sys.argv[4]
+
+        configure_logger(logger=logger,
+                         log_level=LOG_LEVEL,
+                         log_format=DOCKER_ID_ROOT_LOG_FORMAT)
+        configure_logger(logger=pycalico_logger,
+                         log_level=LOG_LEVEL,
+                         log_format=DOCKER_ID_LOG_FORMAT)
+
+        logger = logging.LoggerAdapter(logger, {'docker_id': str(docker_id)[:12]})
+        logger = logging.LoggerAdapter(pycalico_logger, {'docker_id': str(docker_id)[:12]})
+    else:
+        namespace = pod_name = docker_id = None
+        configure_logger(logger=logger,
```

Comment:
I think this might add a second stdout and file handler to the logger, rather than replace the existing handlers. Can you check if this produces doubled logs?